superset/charts/api.py

```diff
@@ -549,13 +549,18 @@ def data(self) -> Response:
             500:
               $ref: '#/components/responses/500'
         """
+        json_body = None
         if request.is_json:
             json_body = request.json
         elif request.form.get("form_data"):
             # CSV export submits regular form data
-            json_body = json.loads(request.form["form_data"])
-        else:
-            return self.response_400(message="Request is not JSON")
+            try:
+                json_body = json.loads(request.form["form_data"])
+            except (TypeError, json.JSONDecodeError):
+                json_body = None
```

Comment:
Note for future: can probably replace this line with `pass` since you already defined `json_body` as `None` at the beginning.